Importing python libraries to be used:

In [45]:
import pandas as pd
import numpy as np

There are 8 different csv files regarding each year. 
<br>To make full analysis throughout 2015-2022, I am going to join each file into main dataframe using 'for in' loop. 

In [46]:
#names of csv source files
years = ['2015', '2016', '2017', '2018', '2019', '2020', '2021','2022']

In [47]:
main_df = pd.DataFrame(pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/2015.csv"))
main_df['Year'] = '2015'
for i in range(1, len(years)):
    sub_df = pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/"+years[i]+r".csv")
    sub_df['Year'] = years[i]
    main_df = pd.concat([main_df,sub_df],axis=0)
    main_df = main_df.reset_index(drop=True)
main_df.head()


Country          Region  Happiness Rank  Happiness Score  \
0  Switzerland  Western Europe             1.0            7.587   
1      Iceland  Western Europe             2.0            7.561   
2      Denmark  Western Europe             3.0            7.527   
3       Norway  Western Europe             4.0            7.522   
4       Canada   North America             5.0            7.427   

   Standard Error  Economy (GDP per Capita)   Family  \
0         0.03411                   1.39651  1.34951   
1         0.04884                   1.30232  1.40223   
2         0.03328                   1.32548  1.36058   
3         0.03880                   1.45900  1.33095   
4         0.03553                   1.32629  1.32261   

   Health (Life Expectancy)  Freedom  Trust (Government Corruption)  ...  \
0                   0.94143  0.66557                        0.41978  ...   
1                   0.94784  0.62877                        0.14145  ...   
2                   0.87464  0.64938                        0.48357  ...   
3                   0.88521  0.66973                        0.36503  ...   
4                   0.90563  0.63297                        0.32957  ...   

   Explained by: Freedom to make life choices  Explained by: Generosity  \
0                                         NaN                       NaN   
1                                         NaN                       NaN   
2                                         NaN                       NaN   
3                                         NaN                       NaN   
4                                         NaN                       NaN   

  Explained by: Perceptions of corruption  Dystopia + residual  RANK  \
0                                     NaN                  NaN   NaN   
1                                     NaN                  NaN   NaN   
2                                     NaN                  NaN   NaN   
3                                     NaN                  NaN   NaN   
4                                     NaN                  NaN   NaN   

   Happiness score  Whisker-high  Whisker-low  Dystopia (1.83) + residual  \
0              NaN           NaN          NaN                         NaN   
1              NaN           NaN          NaN                         NaN   
2              NaN           NaN          NaN                         NaN   
3              NaN           NaN          NaN                         NaN   
4              NaN           NaN          NaN                         NaN   

   Explained by: GDP per capita  
0                           NaN  
1                           NaN  
2                           NaN  
3                           NaN  
4                           NaN  

[5 rows x 57 columns]

Let's see what the dataframe looks like:

In [48]:
main_df.shape

(1231, 57)

The df has 1231 rows and 57 columns. 
<br>The number of columns isn't expected. This might mean that the attributes of each subset of data have different naming so they were added as separate columns.
<br>Let's have a look at the columns:

In [64]:
#how many columns each csv has
column_check = {}
data = pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/"+years[i]+r".csv")

column_check = {years[i] : data.shape[1] for i in range(0, len(years))} #using 1st index of attribute 'shape' which is number of columns
print(column_check)


{'2015': 12, '2016': 12, '2017': 12, '2018': 12, '2019': 12, '2020': 12, '2021': 12, '2022': 12}


In [50]:
#how the columns are named
main_df.columns.sort_values()


Index(['Country', 'Country name', 'Country or region',
       'Dystopia (1.83) + residual', 'Dystopia + residual',
       'Dystopia Residual', 'Dystopia.Residual', 'Economy (GDP per Capita)',
       'Economy..GDP.per.Capita.',
       'Explained by: Freedom to make life choices',
       'Explained by: GDP per capita', 'Explained by: Generosity',
       'Explained by: Healthy life expectancy',
       'Explained by: Log GDP per capita',
       'Explained by: Perceptions of corruption',
       'Explained by: Social support', 'Family', 'Freedom',
       'Freedom to make life choices', 'Freedom to make life choices_',
       'GDP per capita', 'Generosity', 'Generosity_', 'Happiness Rank',
       'Happiness Score', 'Happiness score', 'Happiness.Rank',
       'Happiness.Score', 'Health (Life Expectancy)',
       'Health..Life.Expectancy.', 'Healthy life expectancy',
       'Healthy life expectancy_', 'Ladder score', 'Ladder score in Dystopia',
       'Logged GDP per capita', 'Lower Confidence 

There is significant redundancy in the columns.
<br>For example, information about countries is split into three columns:
* Country
* Country name
* Country or region
<br> The files have between 9-20 columns. Let's have deeper look into 2020 and 2021 as they have much more columns than the rest of subsets:


In [51]:

for i in range(5, 6): #(5,6) are indexes of years list: 2020 and 2021 
    df_20_21 = pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/"+years[i]+r".csv")
df_20_21.columns

Index(['Country name', 'Regional indicator', 'Ladder score',
       'Standard error of ladder score', 'upperwhisker', 'lowerwhisker',
       'Logged GDP per capita', 'Social support_', 'Healthy life expectancy_',
       'Freedom to make life choices_', 'Generosity_',
       'Perceptions of corruption_', 'Ladder score in Dystopia',
       'Explained by: Log GDP per capita', 'Explained by: Social support',
       'Explained by: Healthy life expectancy',
       'Explained by: Freedom to make life choices',
       'Explained by: Generosity', 'Explained by: Perceptions of corruption',
       'Dystopia + residual'],
      dtype='object')

As suspected, these years contain duplicated columns of six happiness factors. Let's compare ranges of these factors:

In [52]:
df_20_21[['Logged GDP per capita',
'Explained by: Log GDP per capita',
'Social support_',
'Explained by: Social support',
'Healthy life expectancy_',
 'Explained by: Healthy life expectancy',
 'Freedom to make life choices_',
 'Explained by: Freedom to make life choices',
 'Generosity_','Explained by: Generosity',
 'Perceptions of corruption_','Explained by: Perceptions of corruption']].describe()

Logged GDP per capita  Explained by: Log GDP per capita  \
count             153.000000                        153.000000   
mean                9.295706                          0.868771   
std                 1.201588                          0.372416   
min                 6.492642                          0.000000   
25%                 8.350645                          0.575862   
50%                 9.456313                          0.918549   
75%                10.265124                          1.169229   
max                11.450681                          1.536676   

       Social support_  Explained by: Social support  \
count       153.000000                    153.000000   
mean          0.808721                      1.155607   
std           0.121453                      0.286866   
min           0.319460                      0.000000   
25%           0.737217                      0.986718   
50%           0.829204                      1.203987   
75%           0.906747                      1.387139   
max           0.974670                      1.547567   

       Healthy life expectancy_  Explained by: Healthy life expectancy  \
count                153.000000                             153.000000   
mean                  64.445529                               0.692869   
std                    7.057848                               0.254094   
min                   45.200001                               0.000000   
25%                   58.961712                               0.495443   
50%                   66.305145                               0.759818   
75%                   69.289192                               0.867249   
max                   76.804581                               1.137814   

       Freedom to make life choices_  \
count                     153.000000   
mean                        0.783360   
std                         0.117786   
min                         0.396573   
25%                         0.714839   
50%                         0.799805   
75%                         0.877709   
max                         0.974998   

       Explained by: Freedom to make life choices  Generosity_  \
count                                  153.000000   153.000000   
mean                                     0.463583    -0.014568   
std                                      0.141172     0.151809   
min                                      0.000000    -0.300907   
25%                                      0.381457    -0.127015   
50%                                      0.483293    -0.033665   
75%                                      0.576665     0.085429   
max                                      0.693270     0.560664   

       Explained by: Generosity  Perceptions of corruption_  \
count                153.000000                  153.000000   
mean                   0.189375                    0.733120   
std                    0.100401                    0.175172   
min                    0.000000                    0.109784   
25%                    0.115006                    0.683019   
50%                    0.176745                    0.783122   
75%                    0.255510                    0.849151   
max                    0.569814                    0.935585   

       Explained by: Perceptions of corruption  
count                               153.000000  
mean                                  0.130718  
std                                   0.113097  
min                                   0.000000  
25%                                   0.055805  
50%                                   0.098435  
75%                                   0.163064  
max                                   0.533162

According to the report documentation, compared columns on the right are modified data used as final output in all other data files. To avoid discrepancies in data ranges, I will use columns to the right.
<br>Our final column adjustment looks as below:

In [53]:
correct_names = {
'Country name' : 'Country',
'Country or region' : 'Country',
'Dystopia (1.83) + residual' : 'Dystopia Residual',
'Dystopia + residual' : 'Dystopia Residual',
'Dystopia.Residual' : 'Dystopia Residual',
'Economy (GDP per Capita)' : 'GDP per Capita',
'Economy..GDP.per.Capita.' : 'GDP per Capita',
'Explained by: Freedom to make life choices' : 'Freedom to Make Life Choices',
'Explained by: GDP per capita' : 'GDP per Capita',
'Explained by: Generosity' : "Generosity",
'Explained by: Healthy life expectancy' : 'Healthy Life Expectancy',
'Explained by: Log GDP per capita' : 'GDP per Capita',
'Explained by: Perceptions of corruption' : 'Corruption Perception',
'Explained by: Social support' : 'Social Support',
'Family' : 'Social Support',
'Freedom' : 'Freedom to Make Life Choices',
'Freedom to make life choices' : 'Freedom to Make Life Choices',
'GDP per capita' : 'GDP per Capita',
'Happiness score' : 'Happiness Score',
'Happiness.Rank' : 'Happiness Rank',
'Happiness.Score' : 'Happiness Score',
'Health (Life Expectancy)' : 'Healthy Life Expectancy',
'Health..Life.Expectancy.' : 'Healthy Life Expectancy',
'Healthy life expectancy' : 'Healthy Life Expectancy',
'Ladder score' : 'Happiness Score',
'Overall rank' : 'Happiness Rank',
'Perceptions of corruption' : 'Corruption Perception',
'RANK' : 'Happiness Rank',
'Regional indicator' : 'Region',
'Score' : 'Happiness Score',
'Social support' : 'Social Support',
'Trust (Government Corruption)' : 'Corruption Perception',
'Trust..Government.Corruption.' : 'Corruption Perception'
}

In [54]:
main_df = main_df.rename(columns=correct_names)
main_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1231 entries, 0 to 1230
Data columns (total 57 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Country                         617 non-null    object 
 1   Region                          315 non-null    object 
 2   Happiness Rank                  315 non-null    float64
 3   Happiness Score                 315 non-null    float64
 4   Standard Error                  158 non-null    float64
 5   GDP per Capita                  315 non-null    float64
 6   Social Support                  470 non-null    float64
 7   Healthy Life Expectancy         315 non-null    float64
 8   Freedom to Make Life Choices    470 non-null    float64
 9   Corruption Perception           315 non-null    float64
 10  Generosity                      782 non-null    float64
 11  Dystopia Residual               315 non-null    float64
 12  Year                            12

In [55]:

main_df = pd.DataFrame(pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/2015.csv"))
main_df = main_df.rename(columns=correct_names)
main_df['Year'] = '2015'

for i in range(1, len(years)):
    sub_df = pd.read_csv(r"C:/Users/Agata/Documents/GitHub/Happiness-Report/source_files/"+years[i]+r".csv")
    sub_df = sub_df.rename(columns=correct_names)
    sub_df['Year'] = years[i]
    main_df = pd.concat([main_df,sub_df],axis=0)
    main_df = main_df.reset_index(drop=True)
    sub_df.to_csv(r"C:/Users/Agata/Documents/"+years[i]+r".csv", index=True)

main_df['Year'] = main_df['Year'].astype(int)
main_df #it allows us to see both head and tail of the dataframe


Country          Region  Happiness Rank  Happiness Score  \
0     Switzerland  Western Europe             1.0            7.587   
1         Iceland  Western Europe             2.0            7.561   
2         Denmark  Western Europe             3.0            7.527   
3          Norway  Western Europe             4.0            7.522   
4          Canada   North America             5.0            7.427   
...           ...             ...             ...              ...   
1226      Rwanda*             NaN           143.0            3.268   
1227     Zimbabwe             NaN           144.0            2.995   
1228      Lebanon             NaN           145.0            2.955   
1229  Afghanistan             NaN           146.0            2.404   
1230           xx             NaN           147.0              NaN   

      Standard Error  GDP per Capita  Social Support  Healthy Life Expectancy  \
0            0.03411         1.39651         1.34951                  0.94143   
1            0.04884         1.30232         1.40223                  0.94784   
2            0.03328         1.32548         1.36058                  0.87464   
3            0.03880         1.45900         1.33095                  0.88521   
4            0.03553         1.32629         1.32261                  0.90563   
...              ...             ...             ...                      ...   
1226             NaN         0.00100         0.00000                  0.46200   
1227             NaN         0.00100         0.00100                  0.27000   
1228             NaN         1.39200         0.00000                  0.63100   
1229             NaN         0.00100         0.00000                  0.28900   
1230             NaN             NaN             NaN                      NaN   

      Freedom to Make Life Choices  Corruption Perception  ...  lowerwhisker  \
0                          0.66557                0.41978  ...           NaN   
1                          0.62877                0.14145  ...           NaN   
2                          0.64938                0.48357  ...           NaN   
3                          0.66973                0.36503  ...           NaN   
4                          0.63297                0.32957  ...           NaN   
...                            ...                    ...  ...           ...   
1226                       0.62100                0.54400  ...           NaN   
1227                       0.32900                0.10500  ...           NaN   
1228                       0.10300                0.03400  ...           NaN   
1229                       0.00000                0.00500  ...           NaN   
1230                           NaN                    NaN  ...           NaN   

      Logged GDP per capita  Social support_  Healthy life expectancy_  \
0                       NaN              NaN                       NaN   
1                       NaN              NaN                       NaN   
2                       NaN              NaN                       NaN   
3                       NaN              NaN                       NaN   
4                       NaN              NaN                       NaN   
...                     ...              ...                       ...   
1226                    NaN              NaN                       NaN   
1227                    NaN              NaN                       NaN   
1228                    NaN              NaN                       NaN   
1229                    NaN              NaN                       NaN   
1230                    NaN              NaN                       NaN   

      Freedom to make life choices_  Generosity_  Perceptions of corruption_  \
0                               NaN          NaN                         NaN   
1                               NaN          NaN                         NaN   
2                               NaN          NaN                         NaN   
3                         

In [56]:
main_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1231 entries, 0 to 1230
Data columns (total 29 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Country                         1231 non-null   object 
 1   Region                          617 non-null    object 
 2   Happiness Rank                  929 non-null    float64
 3   Happiness Score                 1230 non-null   float64
 4   Standard Error                  158 non-null    float64
 5   GDP per Capita                  1230 non-null   float64
 6   Social Support                  1230 non-null   float64
 7   Healthy Life Expectancy         1230 non-null   float64
 8   Freedom to Make Life Choices    1230 non-null   float64
 9   Corruption Perception           1229 non-null   float64
 10  Generosity                      1230 non-null   float64
 11  Dystopia Residual               918 non-null    float64
 12  Year                            12

In [57]:
cut_df = main_df.loc[:,:'Year']
cut_df = cut_df.drop(columns='Standard Error')
cut_df = cut_df.drop(axis=0, index=1230)
cut_df['Country'] = cut_df['Country'].str.replace('*','')
cut_df.loc[cut_df.Year.isin([2020,2021])]

C:\Users\Agata\AppData\Local\Temp\ipykernel_14276\2437194554.py:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  cut_df['Country'] = cut_df['Country'].str.replace('*','')


Country              Region  Happiness Rank  Happiness Score  \
782       Finland      Western Europe             NaN           7.8087   
783       Denmark      Western Europe             NaN           7.6456   
784   Switzerland      Western Europe             NaN           7.5599   
785       Iceland      Western Europe             NaN           7.5045   
786        Norway      Western Europe             NaN           7.4880   
...           ...                 ...             ...              ...   
1079      Lesotho  Sub-Saharan Africa             NaN           3.5120   
1080     Botswana  Sub-Saharan Africa             NaN           3.4670   
1081       Rwanda  Sub-Saharan Africa             NaN           3.4150   
1082     Zimbabwe  Sub-Saharan Africa             NaN           3.1450   
1083  Afghanistan          South Asia             NaN           2.5230   

      GDP per Capita  Social Support  Healthy Life Expectancy  \
782         1.285190        1.499526                 0.961271   
783         1.326949        1.503449                 0.979333   
784         1.390774        1.472403                 1.040533   
785         1.326502        1.547567                 1.000843   
786         1.424207        1.495173                 1.008072   
...              ...             ...                      ...   
1079        0.451000        0.731000                 0.007000   
1080        1.099000        0.724000                 0.340000   
1081        0.364000        0.202000                 0.407000   
1082        0.457000        0.649000                 0.243000   
1083        0.370000        0.000000                 0.126000   

      Freedom to Make Life Choices  Corruption Perception  Generosity  \
782                       0.662317               0.477857    0.159670   
783                       0.665040               0.495260    0.242793   
784                       0.628954               0.407946    0.269056   
785                       0.661981               0.144541    0.362330   
786                       0.670201               0.434101    0.287985   
...                            ...                    ...         ...   
1079                      0.405000               0.015000    0.103000   
1080                      0.539000               0.088000    0.027000   
1081                      0.627000               0.493000    0.227000   
1082                      0.359000               0.075000    0.157000   
1083                      0.000000               0.010000    0.122000   

      Dystopia Residual  Year  
782            2.762835  2020  
783            2.432741  2020  
784            2.350267  2020  
785            2.460688  2020  
786            2.168266  2020  
...                 ...   ...  
1079           1.800000  2021  
1080           0.648000  2021  
1081           1.095000  2021  
1082           1.205000  2021  
1083           1.895000  2021  

[302 rows x 12 columns]

In [58]:
rank = cut_df[cut_df.Year.isin([2020,2021])]
rank['Happiness Rank'] = rank.groupby('Year')['Happiness Score'].rank('dense',ascending=False)
#rank.to_csv(r'C:/Users/Agata/Desktop/test_rank.csv', index=True)
cut_df.loc[782:1083,'Happiness Rank'] = rank['Happiness Rank']
cut_df[cut_df.Year.isin([2020,2021])]

C:\Users\Agata\AppData\Local\Temp\ipykernel_14276\3876616815.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  rank['Happiness Rank'] = rank.groupby('Year')['Happiness Score'].rank('dense',ascending=False)


Country              Region  Happiness Rank  Happiness Score  \
782       Finland      Western Europe             1.0           7.8087   
783       Denmark      Western Europe             2.0           7.6456   
784   Switzerland      Western Europe             3.0           7.5599   
785       Iceland      Western Europe             4.0           7.5045   
786        Norway      Western Europe             5.0           7.4880   
...           ...                 ...             ...              ...   
1079      Lesotho  Sub-Saharan Africa           143.0           3.5120   
1080     Botswana  Sub-Saharan Africa           144.0           3.4670   
1081       Rwanda  Sub-Saharan Africa           145.0           3.4150   
1082     Zimbabwe  Sub-Saharan Africa           146.0           3.1450   
1083  Afghanistan          South Asia           147.0           2.5230   

      GDP per Capita  Social Support  Healthy Life Expectancy  \
782         1.285190        1.499526                 0.961271   
783         1.326949        1.503449                 0.979333   
784         1.390774        1.472403                 1.040533   
785         1.326502        1.547567                 1.000843   
786         1.424207        1.495173                 1.008072   
...              ...             ...                      ...   
1079        0.451000        0.731000                 0.007000   
1080        1.099000        0.724000                 0.340000   
1081        0.364000        0.202000                 0.407000   
1082        0.457000        0.649000                 0.243000   
1083        0.370000        0.000000                 0.126000   

      Freedom to Make Life Choices  Corruption Perception  Generosity  \
782                       0.662317               0.477857    0.159670   
783                       0.665040               0.495260    0.242793   
784                       0.628954               0.407946    0.269056   
785                       0.661981               0.144541    0.362330   
786                       0.670201               0.434101    0.287985   
...                            ...                    ...         ...   
1079                      0.405000               0.015000    0.103000   
1080                      0.539000               0.088000    0.027000   
1081                      0.627000               0.493000    0.227000   
1082                      0.359000               0.075000    0.157000   
1083                      0.000000               0.010000    0.122000   

      Dystopia Residual  Year  
782            2.762835  2020  
783            2.432741  2020  
784            2.350267  2020  
785            2.460688  2020  
786            2.168266  2020  
...                 ...   ...  
1079           1.800000  2021  
1080           0.648000  2021  
1081           1.095000  2021  
1082           1.205000  2021  
1083           1.895000  2021  

[302 rows x 12 columns]

In [59]:
cut_df[cut_df.Year.isin([2016])]

Country                           Region  Happiness Rank  \
158      Denmark                   Western Europe             1.0   
159  Switzerland                   Western Europe             2.0   
160      Iceland                   Western Europe             3.0   
161       Norway                   Western Europe             4.0   
162      Finland                   Western Europe             5.0   
..           ...                              ...             ...   
310        Benin               Sub-Saharan Africa           153.0   
311  Afghanistan                    Southern Asia           154.0   
312         Togo               Sub-Saharan Africa           155.0   
313        Syria  Middle East and Northern Africa           156.0   
314      Burundi               Sub-Saharan Africa           157.0   

     Happiness Score  GDP per Capita  Social Support  Healthy Life Expectancy  \
158            7.526         1.44178         1.16374                  0.79504   
159            7.509         1.52733         1.14524                  0.86303   
160            7.501         1.42666         1.18326                  0.86733   
161            7.498         1.57744         1.12690                  0.79579   
162            7.413         1.40598         1.13464                  0.81091   
..               ...             ...             ...                      ...   
310            3.484         0.39499         0.10419                  0.21028   
311            3.360         0.38227         0.11037                  0.17344   
312            3.303         0.28123         0.00000                  0.24811   
313            3.069         0.74719         0.14866                  0.62994   
314            2.905         0.06831         0.23442                  0.15747   

     Freedom to Make Life Choices  Corruption Perception  Generosity  \
158                       0.57941                0.44453     0.36171   
159                       0.58557                0.41203     0.28083   
160                       0.56624                0.14975     0.47678   
161                       0.59609                0.35776     0.37895   
162                       0.57104                0.41004     0.25492   
..                            ...                    ...         ...   
310                       0.39747                0.06681     0.20180   
311                       0.16430                0.07112     0.31268   
312                       0.34678                0.11587     0.17517   
313                       0.06912                0.17233     0.48397   
314                       0.04320                0.09419     0.20290   

     Dystopia Residual  Year  
158            2.73939  2016  
159            2.69463  2016  
160            2.83137  2016  
161            2.66465  2016  
162            2.82596  2016  
..                 ...   ...  
310            2.10812  2016  
311            2.14558  2016  
312            2.13540  2016  
313            0.81789  2016  
314            2.10404  2016  

[157 rows x 12 columns]

In [60]:
c = cut_df.loc[158:314].Country.tolist()
r = cut_df.loc[158:314].Region.tolist()
region = {c[i] : r[i] for i in range(len(c))}
cut_df['Region'] = cut_df.Country.map(region)
cut_df


Country                           Region  Happiness Rank  \
0     Switzerland                   Western Europe             1.0   
1         Iceland                   Western Europe             2.0   
2         Denmark                   Western Europe             3.0   
3          Norway                   Western Europe             4.0   
4          Canada                    North America             5.0   
...           ...                              ...             ...   
1225     Botswana               Sub-Saharan Africa           142.0   
1226       Rwanda               Sub-Saharan Africa           143.0   
1227     Zimbabwe               Sub-Saharan Africa           144.0   
1228      Lebanon  Middle East and Northern Africa           145.0   
1229  Afghanistan                    Southern Asia           146.0   

      Happiness Score  GDP per Capita  Social Support  \
0               7.587         1.39651         1.34951   
1               7.561         1.30232         1.40223   
2               7.527         1.32548         1.36058   
3               7.522         1.45900         1.33095   
4               7.427         1.32629         1.32261   
...               ...             ...             ...   
1225            3.471         1.50300         0.00100   
1226            3.268         0.00100         0.00000   
1227            2.995         0.00100         0.00100   
1228            2.955         1.39200         0.00000   
1229            2.404         0.00100         0.00000   

      Healthy Life Expectancy  Freedom to Make Life Choices  \
0                     0.94143                       0.66557   
1                     0.94784                       0.62877   
2                     0.87464                       0.64938   
3                     0.88521                       0.66973   
4                     0.90563                       0.63297   
...                       ...                           ...   
1225                  0.28000                       0.57100   
1226                  0.46200                       0.62100   
1227                  0.27000                       0.32900   
1228                  0.63100                       0.10300   
1229                  0.28900                       0.00000   

      Corruption Perception  Generosity  Dystopia Residual  Year  
0                   0.41978     0.29678            2.51738  2015  
1                   0.14145     0.43630            2.70201  2015  
2                   0.48357     0.34139            2.49204  2015  
3                   0.36503     0.34699            2.46531  2015  
4                   0.32957     0.45811            2.45176  2015  
...                     ...         ...                ...   ...  
1225                0.10200     0.01200            0.00000  2022  
1226                0.54400     0.18700            0.00100  2022  
1227                0.10500     0.10600            0.00100  2022  
1228                0.03400     0.08200            0.00000  2022  
1229                0.00500     0.08900            1.26300  2022  

[1230 rows x 12 columns]

In [61]:
cut_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1230 entries, 0 to 1229
Data columns (total 12 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Country                       1230 non-null   object 
 1   Region                        1180 non-null   object 
 2   Happiness Rank                1230 non-null   float64
 3   Happiness Score               1230 non-null   float64
 4   GDP per Capita                1230 non-null   float64
 5   Social Support                1230 non-null   float64
 6   Healthy Life Expectancy       1230 non-null   float64
 7   Freedom to Make Life Choices  1230 non-null   float64
 8   Corruption Perception         1229 non-null   float64
 9   Generosity                    1230 non-null   float64
 10  Dystopia Residual             918 non-null    float64
 11  Year                          1230 non-null   int32  
dtypes: float64(9), int32(1), object(2)
memory usage: 110.6+ KB


In [62]:
cut_df.to_csv(r'C:/Users/Agata/Documents/GitHub/Happiness-Report/cleaned_data.csv', index=False)